lib/open_feature/sdk/provider_initialization_error.rb

```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+module OpenFeature
+  module SDK
+    # Exception raised when a provider fails to initialize during setProviderAndWait
+    #
+    # This exception provides access to both the original error that caused the
+    # initialization failure and the provider instance that failed to initialize.
+    class ProviderInitializationError < StandardError
```

Comment:
This should likely have the error code `PROVIDER_FATAL`.

https://openfeature.dev/specification/types#error-code